packages/ui/src/components/Table/TablePaginationItem.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+    KeyboardArrowLeft,
+    KeyboardArrowRight,
+    KeyboardDoubleArrowLeft,
+    KeyboardDoubleArrowRight,
+    MoreHorizRounded,
+} from '@mui/icons-material';
+import { Button } from '@mui/material';
+import { UsePaginationItem } from '@mui/material/usePagination/usePagination';
+import { FC } from 'react';
+
+interface TablePaginationItemProps {
+    item: UsePaginationItem;
+}
+
+export const TablePaginationItem: FC<TablePaginationItemProps> = ({ item }) => {
+    const { page, type, onClick, disabled, selected } = item;
+
+    if (type === 'start-ellipsis' || type === 'end-ellipsis') {
+        return <MoreHorizRounded sx={{ mr: '0.5rem' }} fontSize="small" />;
+    }
+
+    const getItem = (givenType: string) => {
+        switch (givenType) {
+            case 'page':
+                return page;
+            case 'next':
+                return <KeyboardArrowRight color={disabled ? 'disabled' : 'primary'} />;
```

Comment:
you can have a const `color = disabled ? 'disabled' : 'primary'` before the switch and pass that as props